### PR TITLE
[SPARK-43923][CONNECT][FOLLOWUP]  Correct the message abbreviation

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
1, truncate raw bytes (udf/udtf/local relation) with `MAX_BYTES_SIZE`;
2, pass `maxStringSize` to abbreviate nested messages;
3, minor optimization to avoid temp array creation;

### Why are the changes needed?
1, there is only one place specifying the `maxStringSize`, with value `MAX_STATEMENT_TEXT_SIZE = 65535`. By its name, it is used to truncate the SQL statements which are always strings. No need to affect raw bytes;

2, according to the implementation of `Message.toString`:

https://github.com/protocolbuffers/protobuf/blob/main/java/core/src/main/java/com/google/protobuf/TextFormat.java#L567-L574

the value of bytes fields can be either `ByteString` or `byte[]`, so the two branches should be consistent.

3, `maxStringSize` only affects the top-level string fields, it should also be used in nested messages.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @HyukjinKwon @jdesjean

**Review by HyukjinKwon — APPROVED** [review]

File: connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/ProtoUtils.scala [hunk 1]
```diff
@@ -54,28 +57,28 @@ private[connect] object ProtoUtils {
           if field.getJavaType == FieldDescriptor.JavaType.BYTE_STRING && byteArray != null =>
         val size = byteArray.size
         if (size > MAX_BYTES_SIZE) {
-          val prefix = byteArray.take(MAX_BYTES_SIZE)
-          builder.setField(field, createByteString(prefix, size))
+          builder.setField(
+            field,
+            ByteString
+              .copyFrom(byteArray, 0, MAX_BYTES_SIZE)
```
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/ProtoUtils.scala:63]: It always show the hex string in UI, don't need to take specified charset into account

[the remark this answers is not in the document]
Comment (LuciferYang) [reply connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/ProtoUtils.scala:63]: Got it

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by LuciferYang** [comment]
Merged into master. Thanks @zhengruifeng @HyukjinKwon

**Pull request closed without merging**